src/Compilers/VisualBasic/Portable/BoundTree/Expression.vb

```diff
@@ -374,14 +355,24 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                         argument,
                         Function(argumentValue) New ByRefArgument(If(CUInt(index) < CUInt(parameters.Length), parameters(index), Nothing), DirectCast(argumentValue, BoundByRefArgumentWithCopyBack)))
                 Case Else
-                    ' Apparently the VB bound trees don't encode named arguments, which seems unnecesarily lossy.
                     Return s_argumentMappings.GetValue(
                         argument,
                         Function(argumentValue)
-                            If index >= parameters.Length - 1 AndAlso parameters.Length > 0 AndAlso parameters(parameters.Length - 1).IsParamArray Then
-                                Return New Argument(ArgumentKind.ParamArray, parameters(parameters.Length - 1), argumentValue)
+                            Dim lastParameterIndex = parameters.Length - 1
+                            If index >= lastParameterIndex AndAlso parameters.Length > 0 AndAlso parameters(lastParameterIndex).IsParamArray Then
```

Comment:
Should probably also check that the argument is an array creation.

---
In reply to: [110500357](https://github.com/dotnet/roslyn/pull/18494#discussion_r110500357) [](ancestors = 110500357)